### ISSUE TITLE
🐛 Bug Report: List template tasks page not working in `scaffolder/next`

### ISSUE THREAD

**Issue by pcgqueiroz** [backstage/backstage#16838]
### 📜 Description

When trying to open the _List Template Tasks_ (component `<ListTaskPageContent />`), when using the new `scaffolder/next` version, occurs an error and the page is not displayed.

### 👍 Expected behavior

The page should open.

### 👎 Actual Behavior with Screenshots

![image](https://user-images.githubusercontent.com/42942793/224938210-0f313625-4ce8-4a00-99a6-05c6d9f47bec.png)

### 👟 Reproduction steps

Go to the Create Template Main Page and click on the menu option `Task List`.

![image](https://user-images.githubusercontent.com/42942793/224938536-1487481d-4928-4ae0-a30b-2d1862c645d8.png)

### 📃 Provide the context for the Bug.

I am testing the new `scaffolder/next` component.

These changes have fixed the problem:

https://github.com/backstage/backstage/blob/master/plugins/scaffolder/src/components/ListTasksPage/ListTasksPage.tsx#L42-L52

```diff
-import { rootRouteRef } from '../../routes';
+import { nextRouteRef } from '../../next';

export interface MyTaskPageProps {
  initiallySelectedFilter?: 'owned' | 'all';
}

const ListTaskPageContent = (props: MyTaskPageProps) => {
  const { initiallySelectedFilter = 'owned' } = props;

  const scaffolderApi = useApi(scaffolderApiRef);
-  const rootLink = useRouteRef(rootRouteRef);
+  const rootLink = useRouteRef(nextRouteRef);
```

### 🖥️ Your Environment

OS:   Linux 5.19.0-35-generic - linux/x64
node: v16.16.0
yarn: 1.22.19
cli:  0.22.3 (installed)
backstage:  1.11.0

Dependencies:
  @backstage/app-defaults                          1.2.0
  @backstage/backend-app-api                       0.4.0
  @backstage/backend-common                        0.18.2
  @backstage/backend-dev-utils                     0.1.0
  @backstage/backend-plugin-api                    0.4.0
  @backstage/backend-tasks                         0.4.3
  @backstage/catalog-client                        1.3.1
  @backstage/catalog-model                         1.2.0
  @backstage/cli-common                            0.1.11
  @backstage/cli                                   0.22.3
  @backstage/config-loader                         1.1.8
  @backstage/config                                1.0.6
  @backstage/core-app-api                          1.5.0
  @backstage/core-components                       0.12.4
  @backstage/core-plugin-api                       1.4.0
  @backstage/dev-utils                             1.0.12
  @backstage/errors                                1.1.4
  @backstage/eslint-plugin                         0.1.1
  @backstage/integration-aws-node                  0.1.1
  @backstage/integration-react                     1.1.10
  @backstage/integration                           1.4.2
  @backstage/plugin-api-docs                       0.9.0
  @backstage/plugin-auth-backend                   0.18.0
  @backstage/plugin-auth-node                      0.2.11
  @backstage/plugin-badges-backend                 0.1.36
  @backstage/plugin-badges                         0.2.39
  @backstage/plugin-catalog-backend-module-github  0.2.5
  @backstage/plugin-catalog-backend                1.7.2
  @backstage/plugin-catalog-common                 1.0.11
  @backstage/plugin-catalog-graph                  0.2.27
  @backstage/plugin-catalog-import                 0.9.5
  @backstage/plugin-catalog-node                   1.3.3
  @backstage/plugin-catalog-react                  1.3.0
  @backstage/plugin-catalog                        1.8.0
  @backstage/plugin-events-node                    0.2.3
  @backstage/plugin-explore-backend                0.0.4
  @backstage/plugin-explore-common                 0.0.1
  @backstage/plugin-explore-react                  0.0.26
  @backstage/plugin-explore                        0.4.0
  @backstage/plugin-github-actions                 0.5.15
  @backstage/plugin-home                           0.4.31
  @backstage/plugin-jenkins-backend                0.1.32
  @backstage/plugin-jenkins-common                 0.1.13
  @backstage/plugin-jenkins                        0.7.14
  @backstage/plugin-org                            0.6.5
  @backstage/plugin-pagerduty                      0.5.8
  @backstage/plugin-permission-backend             0.5.17
  @backstage/plugin-permission-common              0.7.3
  @backstage/plugin-permission-node                0.7.5
  @backstage/plugin-permission-react               0.4.10
  @backstage/plugin-playlist-common                0.1.4
  @backstage/plugin-playlist                       0.1.6
  @backstage/plugin-proxy-backend                  0.2.36
  @backstage/plugin-scaffolder-backend             1.11.0
  @backstage/plugin-scaffolder-common              1.2.5
  @backstage/plugin-scaffolder-node                0.1.0
  @backstage/plugin-scaffolder-react               1.1.0
  @backstage/plugin-scaffolder                     1.11.0
  @backstage/plugin-search-backend-module-pg       0.5.3
  @backstage/plugin-search-backend-node            1.1.3
  @backstage/plugin-search-backend                 1.2.3
  @backstage/plugin-search-common                  1.2.1
  @backstage/plugin-search-react                   1.5.0
  @backstage/plugin-search                         1.1.0
  @backstage/plugin-sonarqube-backend              0.1.7
  @backstage/plugin-sonarqube-react                0.1.3
  @backstage/plugin-sonarqube                      0.6.4
  @backstage/plugin-tech-radar                     0.6.1
  @backstage/plugin-techdocs-backend               1.5.3
  @backstage/plugin-techdocs-module-addons-contrib 1.0.10
  @backstage/plugin-techdocs-node                  1.5.0
  @backstage/plugin-techdocs-react                 1.1.3
  @backstage/plugin-techdocs                       1.5.0
  @backstage/plugin-todo                           0.2.17
  @backstage/plugin-user-settings-backend          0.1.6
  @backstage/plugin-user-settings                  0.7.0
  @backstage/release-manifests                     0.0.8
  @backstage/test-utils                            1.2.5
  @backstage/theme                                 0.2.17
  @backstage/types                                 1.0.2
  @backstage/version-bridge                        1.0.3

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by benjdlambert** [comment]
Hey :wave:

This should have been fixed in #16942 will be released in the `next` release next tuesday!

**Comment by benjdlambert** [comment]
Closing has been fixed in 1.3.0-next.0

**Comment by benjdlambert** [comment]
@brahimk please raise another feature request. This is a bug ticket, and not related to your issue I believe.
